### DISCUSSION TITLE
[Question] cpp2 support for brace initializer list for a container?

### DISCUSSION THREAD

**Discussion by guybrush77** [hsutter/cppfront#678] [Q&A] [answered]
Does cpp2 support brace initializer list for a container? For example, something like:

```
m: std::map<int, int>;

operator=: (out this) = {
    m = ( (1, 2), (3, 4) );
}
```

**Comment by JohelEGP** [comment, marked as answer]
No, see #542.
You need to explicitly specify the elements.
In this case, it'd be `std::pair`.
```Cpp2
    m = ( std::pair(1, 2), std::pair(3, 4) );
```

**Reply by guybrush77** [reply]
Thanks for the prompt reply. It's interesting... Would be nice to improve initialization, though it looks tricky to me.

**Comment by guybrush77** [comment]
Looks like I can't initialize in the constructor:

```
foo: type = {
    m: std::map<int, int>;
    operator=: (out this, bar: int) = { m = ( std::pair(1, 2), std::pair(3, 4) ); }
}

main: () = {
    f := foo(5);
}
```

I get this error using MSVC:
```
repro.cpp2(7): error C2679: binary '=': no operator found which takes a right-hand operand of type 'std::pair<int,int>' (or there is no acceptable conversion)
```

If I remove the `bar` parameter then it works.

**Reply by JohelEGP** [reply]
Looks like a bug, as the generated assignment is (<https://cpp2.godbolt.org/z/Yc9Go56Ej>):
```C++
    auto foo::operator=(cpp2::in<int> bar) -> foo& {
                                        m = std::pair(1, 2), std::pair(3, 4);
                                        return *this;
    }
```

**Comment by guybrush77** [comment]
I filed the bug here: https://github.com/hsutter/cppfront/issues/679
